### PR TITLE
MGDAPI-3498 Updating prometheus alert queries

### PR DESCRIPTION
**Pull request by bryankeane0**
# Issue link
JIRA: https://issues.redhat.com/browse/MGDAPI-3498

# What
Changed prometheus alert queries which many were misfiring. Removed use of flakey `absent` which caused occasional false positives. 

# Verification steps
All of the changes are similar enough so testing one alert effectively tests all changes.

1. Install RHOAM with the changes from this PR.
2. Scale down `threescale-operator-controller-manager` deployment to 0.
3. Scale down `backend-worker` deployment config to 0 and wait 5 minutes.
4. Check prometheus alerts in `Networking` > `Routes` > `Prometheus`
5. 5 minutes after scaling down `backend-worker`, confirm that the `ThreescaleBackendWorkerPod` alert is firing.
6. Scale back up the deployment and deployment-config, wait until the pods launch and check alert stops firing.

### CODE REVIEW

**Comment by Patryk-Stefanski** [comment]
Verified that ThreescaleBackendWorkerPod alert is firing after scaling down controller-manager and backend worker, after scaling back up the alert stops firing.
/lgtm

**Comment by Patryk-Stefanski** [comment]
Verified that ThreescaleBackendWorkerPod alert is firing after scaling down controller-manager and backend worker, after scaling back up the alert stops firing.
/lgtm

**Comment by laurafitzgerald** [comment]
Are the errors in prometheus pod now gone after this change?

**Comment by bryankeane0** [comment]
@laurafitzgerald that's my fault, I forgot to include that in the verification steps. My initial tests showed no errors showing in prometheus logs

File: pkg/products/observability/prometheusRules.go [hunk 1]
```diff
@@ -50,16 +50,6 @@ func (r *Reconciler) newAlertsReconciler(logger l.Logger, installType string) re
 						},
 						Expr: intstr.FromString("(time() - (max( kube_job_status_start_time * ON(job_name) GROUP_RIGHT() kube_job_labels{label_monitoring_key='middleware'} ) BY (job_name, label_cronjob_name) == ON(label_cronjob_name) GROUP_LEFT() max( kube_job_status_start_time * ON(job_name) GROUP_RIGHT() kube_job_labels{label_monitoring_key='middleware'} ) BY (label_cronjob_name))) > 60*60*25"),
 					},
-					{
-						Alert: "CronJobsFailed",
```
Comment (laurafitzgerald) [inline pkg/products/observability/prometheusRules.go:54]: There may be a test update if this alert is checked for being present. but it doesn't need to block this pr.

**Comment by laurafitzgerald** [comment]
@bryankeane0 there is a test update required after the removal of the CronJob alert. One of the e2e tests checks that all alerts are present. 
The change should be made here -> https://github.com/integr8ly/integreatly-operator/blob/master/test/common/alerts_exist.go#L517

**Pull request merged**